docxcompose/composer.py

```diff
@@ -556,13 +557,43 @@ def renumber_bookmarks(self):
             bookmark_id += 1
 
     def renumber_docpr_ids(self):
+        # Ensure that non-visual drawing properties have a unique id
         doc_prs = xpath(
             self.doc.element.body, './/wp:docPr')
         doc_pr_id = 1
         for doc_pr in doc_prs:
             doc_pr.id = doc_pr_id
             doc_pr_id += 1
 
+        parts = [
+            rel.target_part for rel in self.doc.part.rels.values()
+            if rel.reltype in [RT.HEADER, RT.FOOTER, ]
+        ]
+        for part in parts:
+            doc_prs = xpath(part.element, './/wp:docPr')
+            for doc_pr in doc_prs:
+                doc_pr.id = doc_pr_id
+                doc_pr_id += 1
+
+    def renumber_nvpicpr_ids(self):
+        # Ensure that non-visual image properties have a unique id
+        c_nv_prs = xpath(
+            self.doc.element.body, './/pic:cNvPr')
+        c_nv_pr_id = 1
+        for c_nv_pr in c_nv_prs:
```

Comment:
you could use `enumerate(c_nv_prs, 1)` to make it a bit more "pythonic"

Reply:
Be aware that the counter continues when renumbering ids in headers and footers to have unique ids across parts.